### PR TITLE
(#1850986) Add support for alternative interface names

### PR DESCRIPTION
**Pull request by dtardon**

### CODE REVIEW

File: src/network/networkd-network-gperf.gperf [hunk 1]
```diff
@@ -24,7 +24,7 @@ Match.MACAddress,                       config_parse_hwaddrs,
 Match.Path,                             config_parse_strv,                              0,                             offsetof(Network, match_path)
 Match.Driver,                           config_parse_strv,                              0,                             offsetof(Network, match_driver)
 Match.Type,                             config_parse_strv,                              0,                             offsetof(Network, match_type)
-Match.Name,                             config_parse_ifnames,                           0,                             offsetof(Network, match_name)
+Match.Name,                             config_parse_match_ifnames,                           0,                             offsetof(Network, match_name)
```
Comment (msekletar) [inline src/network/networkd-network-gperf.gperf:27]: nitpick, indentation is off.

**Review by msekletar — APPROVED** [review]
LGTM (minor nitpick regarding indentation, but that is not important enough to warrant respin).

**Review by mrc0mmand — CHANGES_REQUESTED** [review]
@dtardon please backport https://github.com/systemd/systemd/pull/16501 as well, otherwise assigning altnames might (and does) fail with -EEXIST.

**Comment by dtardon** [comment]
> @dtardon please backport [systemd/systemd#16501](https://github.com/systemd/systemd/pull/16501) as well, otherwise assigning altnames might (and does) fail with -EEXIST.

Done.

**Comment by mrc0mmand** [comment]
This looks relevant:

```
[2021-09-17 12:30:28] [build-stdout] ../src/network/netdev/netdev.c: In function ‘netdev_load_one’:
[2021-09-17 12:30:28] [build-stdout] ../src/network/netdev/netdev.c:639:13: error: too few arguments to function ‘net_match_config’
[2021-09-17 12:30:28] [build-stdout]   639 |         if (net_match_config(NULL, NULL, NULL, NULL, NULL,
[2021-09-17 12:30:28] [build-stdout]       |             ^~~~~~~~~~~~~~~~
[2021-09-17 12:30:28] [build-stdout] In file included from ../src/network/netdev/netdev.c:11:
[2021-09-17 12:30:28] [build-stdout] ../src/libsystemd-network/network-internal.h:17:6: note: declared here
[2021-09-17 12:30:28] [build-stdout]    17 | bool net_match_config(Set *match_mac,
[2021-09-17 12:30:28] [build-stdout]       |      ^~~~~~~~~~~~~~~~
```

**Comment by dtardon** [comment]
> This looks relevant:
> 
> ```
> [2021-09-17 12:30:28] [build-stdout] ../src/network/netdev/netdev.c: In function ‘netdev_load_one’:
> [2021-09-17 12:30:28] [build-stdout] ../src/network/netdev/netdev.c:639:13: error: too few arguments to function ‘net_match_config’
> ```

That file shouldn't even be built, should it? It's part of networkd.

**Comment by mrc0mmand** [comment]
> > This looks relevant:
> > ```
> > [2021-09-17 12:30:28] [build-stdout] ../src/network/netdev/netdev.c: In function ‘netdev_load_one’:
> > [2021-09-17 12:30:28] [build-stdout] ../src/network/netdev/netdev.c:639:13: error: too few arguments to function ‘net_match_config’
> > ```
> 
> That file shouldn't even be built, should it? It's part of networkd.

LGTM doesn't use the RHEL feature switches, so it builds the "default" configuration.

**Comment by dtardon** [comment]
> LGTM doesn't use the RHEL feature switches, so it builds the "default" configuration.

Well, I intentionally dropped most changes in `src/network` because of too many merge conflicts. I didn't--and still don't--want to spend hours or even days fixing build of something we don't even use...

**Comment by mrc0mmand** [comment]
> > LGTM doesn't use the RHEL feature switches, so it builds the "default" configuration.
> 
> Well, I intentionally dropped most changes in `src/network` because of too many merge conflicts. I didn't--and still don't--want to spend hours or even days fixing build of something we don't even use...

Lemme see if I can tweak the LGTM config.

**Comment by mrc0mmand** [comment]
@dtardon could you add a simple RHEL-only patch, which disables networkd by default? I.e.:

```patch
diff --git a/meson_options.txt b/meson_options.txt
index 213079ac15..62a1302f54 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -86,7 +86,7 @@ option('machined', type : 'boolean',
        description : 'install the systemd-machined stack')
 option('portabled', type : 'boolean',
        description : 'install the systemd-portabled stack')
-option('networkd', type : 'boolean',
+option('networkd', type : 'boolean', value : false,
        description : 'install the systemd-networkd stack')
 option('timedated', type : 'boolean',
        description : 'install the systemd-timedated daemon')
```

that should be imo the easiest option here.

**Comment by dtardon** [comment]
LGTM fix continued in #211.

**Pull request merged**